desktop_version/src/DiscordNetwork.c

```diff
@@ -0,0 +1,227 @@
+#define DISCORD_LIBRARY "discord_game_sdk.dll"
+#elif defined(__APPLE__)
+#define DISCORD_LIBRARY "libdiscord_game_sdk.dylib"
+#elif defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__)
+#define DISCORD_LIBRARY "libdiscord_game_sdk.so"
+#else
+#error DISCORD_LIBRARY: Unrecognized platform!
+#endif
+
+
+#define DISCORD_CLIENT_ID 1315544357532729447
+
+// TO TERRY/FLIBIT: You can create your own Discord instance at the Discord Developer Portal. This ID belongs to me, so just be aware that if my account was to get hacked, VVVVVV RPC would too. Use your own!
+
+
+struct DISCORD_application {
+    struct IDiscordCore* core;
+    struct IDiscordActivityManager* activityMan;
+} app;
+
+struct DiscordActivity activity;
+
+bool discordDetected = false;
+bool discordPostInit = false;
+
+
+static void* libHandle = NULL;
+
+static const char* DiscordResult_Strings[] = {
+    "DiscordResult_Ok",
+    "DiscordResult_ServiceUnavailable",
+    "DiscordResult_InvalidVersion",
+    "DiscordResult_LockFailed",
+    "DiscordResult_InternalError",
+    "DiscordResult_InvalidPayload",
+    "DiscordResult_InvalidCommand",
+    "DiscordResult_InvalidPermissions",
+    "DiscordResult_NotFetched",
+    "DiscordResult_NotFound",
+    "DiscordResult_Conflict",
+    "DiscordResult_InvalidSecret",
+    "DiscordResult_InvalidJoinSecret",
+    "DiscordResult_NoEligibleActivity",
+    "DiscordResult_InvalidInvite",
+    "DiscordResult_NotAuthenticated",
+    "DiscordResult_InvalidAccessToken",
+    "DiscordResult_ApplicationMismatch",
+    "DiscordResult_InvalidDataUrl",
+    "DiscordResult_InvalidBase64",
+    "DiscordResult_NotFiltered",
+    "DiscordResult_LobbyFull",
+    "DiscordResult_InvalidLobbySecret",
+    "DiscordResult_InvalidFilename",
+    "DiscordResult_InvalidFileSize",
+    "DiscordResult_InvalidEntitlement",
+    "DiscordResult_NotInstalled",
+    "DiscordResult_NotRunning",
+    "DiscordResult_InsufficientBuffer",
+    "DiscordResult_PurchaseCanceled",
+    "DiscordResult_InvalidGuild",
+    "DiscordResult_InvalidEvent",
+    "DiscordResult_InvalidChannel",
+    "DiscordResult_InvalidOrigin",
+    "DiscordResult_RateLimited",
+    "DiscordResult_OAuth2Error",
+    "DiscordResult_SelectChannelTimeout",
+    "DiscordResult_GetGuildTimeout",
+    "DiscordResult_SelectVoiceForceRequired",
+    "DiscordResult_CaptureShortcutAlreadyListening",
+    "DiscordResult_UnauthorizedForAchievement",
+    "DiscordResult_InvalidGiftCode",
+    "DiscordResult_PurchaseError",
+    "DiscordResult_TransactionAborted",
+    "DiscordResult_DrawingInitFailed"
+};
+
+
+
+#define FUNC_LIST \
+    FOREACH_FUNC(enum EDiscordResult, DiscordCreate, (DiscordVersion, struct DiscordCreateParams*, struct IDiscordCore**))
+
+
+
+#define FOREACH_FUNC(rettype, name, params) static rettype (*name) params = NULL;
+FUNC_LIST
+#undef FOREACH_FUNC
+
+bool DISCORD_REQUIRE(int x) {
+    //vlog_error(DiscordResult_Strings[x]);
```

Comment:
Yes... is there any way we could workaround this?

Reply:
We have `INBOUNDS_ARR(x, DiscordResult_Strings)` which will return true if index `x` is in bounds.

Since `DISCORD_REQUIRE` is (at least in the current state) used for `run_callbacks`, I feel like maybe it should only print once if an error occurs. We have another macro for that: `WHINE_ONCE`/`WHINE_ONCE_ARGS`.

In all... This should work (definitely in the `x != DiscordResult_Ok` branch below):

```cpp
const char* message = "Unknown";
if (INBOUNDS_ARR(x, DiscordResult_Strings))
{
    message = DiscordResult_Strings[x];
}
WHINE_ONCE_ARGS(("Discord error: %s (%d)", message, x));
```

Reply:
@Daaaav I don't think we should bother with this as spitting out inaccurate error codes is just as worse as a segfault.